Gems/SceneProcessing/Code/Source/Generation/Components/MeshOptimizer/MeshBuilderSkinningInfo.cpp

```diff
@@ -25,8 +25,11 @@ namespace AZ::MeshBuilder
     // constructor
     MeshBuilderSkinningInfo::MeshBuilderSkinningInfo(size_t numOrgVertices)
     {
-        mInfluences.SetNumPreCachedElements(4); // TODO: verify if this is the fastest
-        mInfluences.Resize(numOrgVertices);
+        mInfluences.resize(numOrgVertices);
+        for (auto& subArray : mInfluences)
+        {
+            subArray.reserve(4); // TODO: verify if this is the fastest
```

Comment:
4 is our default value, 8 are often used when bone-based facial animation is involved. If there is no decrease in speed for 8, shall we use that as default? Then we prevent allocations for most (close to all) of the cases. Can you also remove the "TODO:"?

Reply:
With this change, we get the default exponential growth of the capacity of the vector. So that means that for vertices in the face that are using 8 influences, those vectors will only grow in capacity once. I know that a lot of characters have higher vertex density in the face, but even so I'd still guess that it'd be around 20% of the character's vertices are in the face. Do we want to double the overall buffer sizes to account for that 20% case? The tradeoff here is that we over-reserve how much memory we'll need up front, and then not end up using it all.

Reply:
I removed the TODO comment

Reply:
Good points and agreed, let's go with 4